packages/arg-parser/src/arg-parser.ts

```diff
@@ -1,125 +1,313 @@
-    'tlsUseSystemCA',
-    'verbose',
-    'version',
-  ],
-  array: ['eval', 'file'],
-  alias: {
-    h: 'help',
-    p: 'password',
-    u: 'username',
-    f: 'file',
-    'build-info': 'buildInfo',
-    json: 'json', // List explicitly here since it can be a boolean or a string
-    browser: 'browser', // ditto
-    oidcDumpTokens: 'oidcDumpTokens', // ditto
-    oidcRedirectUrl: 'oidcRedirectUri', // I'd get this wrong about 50% of the time
-    oidcIDTokenAsAccessToken: 'oidcIdTokenAsAccessToken', // ditto
-  },
-  configuration: {
+export const cliOptionsRegistry = z.registry<CliOptionsRegistryMetadata>();
+
+/**
+ * CLI options schema with metadata attached via registry
+ */
+export const CliOptionsSchema = z
+  .object({
+    // String options
+    apiVersion: z.string().optional(),
+    authenticationDatabase: z.string().optional(),
+    authenticationMechanism: z.string().optional(),
+    awsAccessKeyId: z.string().optional(),
+    awsIamSessionToken: z.string().optional(),
+    awsSecretAccessKey: z.string().optional(),
+    awsSessionToken: z.string().optional(),
+    csfleLibraryPath: z.string().optional(),
+    cryptSharedLibPath: z.string().optional(),
+    db: z.string().optional(),
+    gssapiHostName: z
+      .string()
+      .optional()
+      .register(cliOptionsRegistry, { unsupported: true }),
+    gssapiServiceName: z.string().optional(),
+    sspiHostnameCanonicalization: z.string().optional(),
+    sspiRealmOverride: z.string().optional(),
+    jsContext: z.string().optional(),
+    host: z.string().optional(),
+    keyVaultNamespace: z.string().optional(),
+    kmsURL: z.string().optional(),
+    locale: z.string().optional(),
+    oidcFlows: z.string().optional(),
+    oidcRedirectUri: z
+      .string()
+      .optional()
+      .register(cliOptionsRegistry, {
+        alias: ['oidcRedirectUrl'],
+      }),
+    password: z
+      .string()
+      .optional()
+      .register(cliOptionsRegistry, { alias: ['p'] }),
+    port: z.string().optional(),
+    username: z
+      .string()
+      .optional()
+      .register(cliOptionsRegistry, { alias: ['u'] }),
+
+    // Deprecated SSL options (now TLS)
+    sslPEMKeyFile: z.string().optional().register(cliOptionsRegistry, {
+      deprecationReplacement: 'tlsCertificateKeyFile',
+    }),
+    sslPEMKeyPassword: z.string().optional().register(cliOptionsRegistry, {
+      deprecationReplacement: 'tlsCertificateKeyFilePassword',
+    }),
+    sslCAFile: z.string().optional().register(cliOptionsRegistry, {
+      deprecationReplacement: 'tlsCAFile',
+    }),
+    sslCertificateSelector: z.string().optional().register(cliOptionsRegistry, {
+      deprecationReplacement: 'tlsCertificateSelector',
+    }),
+    sslCRLFile: z.string().optional().register(cliOptionsRegistry, {
+      deprecationReplacement: 'tlsCRLFile',
+    }),
+    sslDisabledProtocols: z.string().optional().register(cliOptionsRegistry, {
+      deprecationReplacement: 'tlsDisabledProtocols',
+    }),
```

Comment:
Wouldn't we typically want to list the non-deprecated versions here and then mention the deprecated aliases in `deprecationReplacement`? Feels a bit backwards to have the deprecated versions at the top level

Reply:
I agree that this feels backwards but in practice it ends up like this:
- Since both i.e. `sslDisabledProtocols` and `tlsDisabledProtocols` are valid to be parsed, they both exist in schema.
- Then, the idea is if someone did use `sslDisabledProtocols`, we can look into the schema for the field to see if this field is deprecated and if it is use the replacement.
- If it was the other way around, we'd need to lookup all fields for their replacements and check if they match our provided fields. i.e. the hashmap would be in reverse of what we're looking for